### PR TITLE
[model] feat: support qwen-image

### PR DESCRIPTION
**Pull request by FoolPlayer**
### What does this PR do?

Support Qwen-Image full training.

### CODE REVIEW

File: scripts/inference/infer_dit.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
+"""Minimal native-PyTorch text-to-image inference for VeOmni-supported DiT models.
```
Comment (Coach257) [inline scripts/inference/infer_dit.py]: Check this: https://github.com/ByteDance-Seed/VeOmni/blob/main/docs/examples/wan2.1_I2V_1.3B.md#6-inference. 
Inference should be launched by diffusers. See https://huggingface.co/Qwen/Qwen-Image for Qwen-Image inference. The ckpt saved by VeOmni should be able to be loaded by diffusers for inference.

File: veomni/data/dummy_dataset.py [hunk 2]
```diff
@@ -374,6 +374,49 @@ def __getitem__(self, index: int) -> Dict[str, Any]:
         ]
 
 
+class QwenImageDataset(Dataset):
```
Comment (Coach257) [inline veomni/data/dummy_dataset.py:377]: To see if we could merge QwenImageDataset and WanT2VDataset to T2VDataset.

File: tests/e2e/test_e2e_parallel.py [hunk 3]
```diff
@@ -233,6 +242,18 @@ def main(
     ),
 ]
 
+qwen_image_dit_test_cases = [
```
Comment (Coach257) [inline tests/e2e/test_e2e_parallel.py:237]: Maybe merge cases with wan_t2v.

File: veomni/models/module_utils.py [hunk 4]
```diff
@@ -153,6 +153,23 @@ def _load_state_dict(weights_path: str, **kwargs) -> List["StateDictIterator"]:
             shard_files, _ = get_checkpoint_shard_files(weights_path, resolved_weight_file, **kwargs)
             return [StateDictIterator(shard_file) for shard_file in shard_files]
 
+        diffusers_transformer_kwargs = {**kwargs, "subfolder": "transformer"}
```
Comment (Coach257) [inline veomni/models/module_utils.py]: Don't need this patch if parse model_path = Qwen/Qwen-Image/transformer
Different transformers path leads to the same condition model path.

File: configs/dit/qwen_image_sft.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
+model:
+  model_path: Qwen/Qwen-Image
```
Comment (Coach257) [inline configs/dit/qwen_image_sft.yaml]: Qwen/Qwen-Image/transformer

File: veomni/models/diffusers/qwen_image/qwen_image_condition/modeling_qwen_image_condition.py [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+                generator=self.generator,
+            ).to(sample_latents.device)
+            timestep = self.scheduler.timesteps[timestep_ids].to(
+                device=sample_latents.device, dtype=sample_latents.dtype
+            )
+            noisy_latents = self.scheduler.scale_noise(sample_latents, timestep, noise)
+            training_target = noise - sample_latents
+
+            packed_conditions["hidden_states"].append(noisy_latents)
+            packed_conditions["timestep"].append(timestep / 1000)
```
[the remark this answers is not in the document]
Comment (Coach257) [reply veomni/models/diffusers/qwen_image/qwen_image_condition/modeling_qwen_image_condition.py:334]: https://github.com/huggingface/diffusers/blob/907c0c2c76e7a24a22e3280ac40f7f2f800a4b01/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L697

timestep / 1000 outside transformer. So /1000 here in conditon model is correct.

File: veomni/data/dummy_dataset.py (hunk 2 above)
Comment (FoolPlayer) [reply veomni/data/dummy_dataset.py:377]: The data shape for wan and qwen-image is different, 5D shape vs 3D shape, maybe seperate datasets is better

**Review by Coach257 — APPROVED** [review]

File: veomni/models/diffusers/qwen_image/qwen_image_condition/modeling_qwen_image_condition.py [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+            prompt_embeds_mask = prompt_embeds_mask.view(batch_size, seq_len)
+            if bool(prompt_embeds_mask.all()):
+                prompt_embeds_mask = None
+
+        return prompt_embeds, prompt_embeds_mask
+
+    def _encode_image_to_latents(self, image) -> tuple[torch.Tensor, list[tuple[int, int, int]]]:
+        image_tensor = self._image_to_tensor(image).to(device=self.vae.device, dtype=self.vae.dtype)
+        posterior: DiagonalGaussianDistribution = self.vae.encode(image_tensor).latent_dist
+        latents = self._normalize_latents(posterior.mode())
```
Comment (Coach257) [inline veomni/models/diffusers/qwen_image/qwen_image_condition/modeling_qwen_image_condition.py]: Do not use mode here. Use posterior.parameters directly. Save [L, 2C] (mean C, logvar C) as image_embeds, and sample latent dist during training. Use `posterior.mode()` reduce diversity.

Comment (Coach257) [reply veomni/models/diffusers/qwen_image/qwen_image_condition/modeling_qwen_image_condition.py]: Example: 
get_condition: https://github.com/ByteDance-Seed/VeOmni/blob/a4ed599119afb21f5e559f15e95635f0edbbc5c6/veomni/models/diffusers/wan_t2v/wan_condition/modeling_wan_condition.py#L99
process_condition: https://github.com/ByteDance-Seed/VeOmni/blob/a4ed599119afb21f5e559f15e95635f0edbbc5c6/veomni/models/diffusers/wan_t2v/wan_condition/modeling_wan_condition.py#L148

**Review by Coach257 — APPROVED** [review]
